### ISSUE TITLE
Zipkin reports UNKNOWN_SERVICE : graphql_request for router spans

### ISSUE THREAD

**Issue by prasek** [apollographql/router#180]
**Describe the bug**
As a follow-on to #149 I'm now seeing Zipkin report `UNKNOWN_SERVICE : graphql_request` for router spans.

The subgraph spans are also offset, so not sure if trace propagation is working correctly.

**To Reproduce**
1. `git clone git@github.com:apollographql/supergraph-demo-fed2.git`
2. `make docker-up-router-otel` - uses [docker-compose.router-otel.yml](https://github.com/apollographql/supergraph-demo-fed2/blob/main/docker-compose.router-otel.yml)
3. `make smoke-router`
4. http://localhost:9411/ -> `Run Query`
5. see `UNKNOWN_SERVICE : graphql_request` for router requests
6. click `Show` and see the offset subgraph spans

**Expected behavior**
1. router to set a default resource/service name so it shows up in the Zipkin `serviceName` field, and ideally allow setting the field via `config.yaml` and include it in the [config docs](https://www.apollographql.com/docs/router/configuration/)
2. subgraph trace spans to align with the parent router spans, like it does for the Gateway in https://github.com/apollographql/supergraph-demo-fed2#tracing-with-open-telemetry

**Output**
![image](https://user-images.githubusercontent.com/284840/143081639-ecd0bb43-5ea2-4380-b9ef-69172f77f038.png)

**Desktop (please complete the following information):**
```
ProductName:	macOS
ProductVersion:	11.6
BuildVersion:	20G165
```

**Comment by Geal** [comment]
it might be related to https://github.com/open-telemetry/opentelemetry-rust/commit/2d37fd197679011030062d5cde6484016b9a1b3c

**Comment by Geal** [comment]
this should be fixed with #210, at least for item 1. For item 2, the subgraph fetches shifting, I have not been able to reproduce yet
